### DISCUSSION TITLE
Goodwe wallbox reverse engineering

### DISCUSSION THREAD

**Discussion by czechdude** [mletenay/home-assistant-goodwe-inverter#212] [Ideas]
Hi, this integration is awesome! 
I just got the GoodWe wallbox GW11K-HCA and I wonder if I could read somehow the data for HA sensors form the interface that the GoodWe offers. So far I have found open only port 80.

Would you give me some tips how to start and maybe create an integration as well? The interface of the wallbox is only available via Sems Portal App - not even the website. Or over Lan/BT with SolarGo....

The web of the wallbox looks like this:
![image](https://github.com/mletenay/home-assistant-goodwe-inverter/assets/601761/dd3671d6-83b4-4e3b-a979-0141e31fd782)

thanks for any suggestions.

**Comment by matyasrr** [comment]
Hello, do you have any update about your work on this?

**Reply by czechdude** [reply]
no, i dont know how to do it.

**Reply by prezervos** [reply]
@czechdude What is please the username for login to the wallbox via ip address? I have found it ... user:admin pass:admin :D

**Reply by czechdude** [reply]
It is admin/admin, but you dont get to control the  Wallbox, just the wifi module inside, just like the wifi/ethernet module in the inverter.

**Reply by TomasOndriska** [reply]
Hi, did anyone manage to integrate it?

**Comment by czechdude** [comment]
I dont think it is possible. The wallbox is closed and reports only to the
mobile sems app over wifi and solargo over bt. thats it. The new version
will be different. I recommend watching this introduction of new Goodwe
Generation 2 products in Czech. https://www.youtube.com/watch?v=VhA0mfugDAM

čt 27. 2. 2025 v 4:05 odesílatel TomasOndriska ***@***.***>
napsal:

**Comment by prezervos** [comment]
Hello, I have manged it and it is working for me. It is based on integration from mletenay - only few modifications were done. But I don't want to publish it as the code is very very ugly and not omptimized and I don't plan to update it as it works and is "enough" for me and I plan to change the wallbox for a smarter one soon. If anyone wants to maintain the code, I can publish it or I can provide zipped folder which you can unpack to custom_components folder...

**Comment by czechdude** [comment]
🙏 please!
čt 27. 2. 2025 v 11:01 odesílatel prezervos ***@***.***>
napsal:

**Reply by prezervos** [reply]
Do you want me to publish the repo, will you continue the work on it? Or zip file only?

**Comment by czechdude** [comment]
We Will definitelly make it a repo, so if you are willing to create it and
maintain it, then create one. But otherwise i can publish it too.
čt 27. 2. 2025 v 11:57 odesílatel prezervos ***@***.***>
napsal:

**Reply by prezervos** [reply]
Made public: https://github.com/prezervos/goodwe-wallbox-sems-home-assistant

**Reply by Sn0w3y** [reply]
[HCA.G2.modbus.file.pdf](https://github.com/user-attachments/files/24949740/HCA.G2.modbus.file.pdf)

you guys know that there is Modbus for the 2nd Gen :D

**Reply by czechdude** [reply]
> [HCA.G2.modbus.file.pdf](https://github.com/user-attachments/files/24949740/HCA.G2.modbus.file.pdf)
> 
> you guys know that there is Modbus :D

Hannes, you know we are talking about first generation, thanks for the file, but no sane person would buy another Goodwe wallbox after experiences with first generation.

**Reply by Sn0w3y** [reply]
> > [HCA.G2.modbus.file.pdf](https://github.com/user-attachments/files/24949740/HCA.G2.modbus.file.pdf)
> > you guys know that there is Modbus :D
> 
> Hannes, you know we are talking about first generation, thanks for the file, but no sane person would buy another Goodwe wallbox after experiences with first generation.

Okay, i changed it :D Thanks for the Info !

**Comment by smh51** [comment]
@Sn0w3y do you know a way to integrate HCA G2 ?

**Reply by Sn0w3y** [reply]
no

**Reply by pedrodivisez** [reply]
@smh51  @Sn0w3y I have G2 wallbox for some time and my copilot has implemented the integration for modbus: try it out and comment https://github.com/pedrodivisez/goodwe-wallbox-sems-home-assistant/tree/gen2-modbus

**Reply by Sn0w3y** [reply]
> @smh51  @Sn0w3y I have G2 wallbox for some time and my copilot has implemented the integration for modbus: try it out and comment https://github.com/pedrodivisez/goodwe-wallbox-sems-home-assistant/tree/gen2-modbus

Does it work? - Use it
Can you test it? - Test it :)

I will not review AI written Code, sorry 😮‍💨

But glad you got it running! :)

**Reply by pedrodivisez** [reply]
> > @smh51  @Sn0w3y I have G2 wallbox for some time and my copilot has implemented the integration for modbus: try it out and comment https://github.com/pedrodivisez/goodwe-wallbox-sems-home-assistant/tree/gen2-modbus
> 
> Does it work? - Use it Can you test it? - Test it :)
> 
> I will not review AI written Code, sorry 😮‍💨
> 
> But glad you got it running! :)

Thanks @Sn0w3y , 
I'm not asking for a review, It is tested and working. I will publish it in the upstream repo for Goodwe Wallbox of @prezervos . I think it is the best code here so far, because noone published nothing till this day. The Cloud SEMS API reliability is rubbish and so the integration cannot be better than the foundations.

**Reply by smh51** [reply]
hey, i've test it quickly. it seems to work. however, i click the toggle "plug & charge" as it was off, and the integration crashed completly. all the sensors became instantly unavailable and i had to force close the integration et reactivate again.

**Reply by pedrodivisez** [reply]
@smh51  thanks for a quick tes, we can continue with issues here https://github.com/pedrodivisez/goodwe-wallbox-sems-home-assistant/issues
I would need you to look for sems errors in logs and turn on debug logs and look for sems in detailed logging, also what model and firmware you have,. also if you enabled Modbus TCP and what invertor model you have and what connections are green in the wallbox solargo.

**Comment by smh51** [comment]
maybe there is a way by using the sems api : https://github.com/TimSoethout/goodwe-sems-home-assistant/discussions/179
